src/api/c++/z3++.h

```diff
@@ -2830,19 +2966,21 @@ namespace z3 {
         Z3_optimize m_opt;
 
     public:
-        class handle {
+        class handle final {
             unsigned m_h;
         public:
             handle(unsigned h): m_h(h) {}
             unsigned h() const { return m_h; }
         };
         optimize(context& c):object(c) { m_opt = Z3_mk_optimize(c); Z3_optimize_inc_ref(c, m_opt); }
-        optimize(optimize& o):object(o)  {
```

Comment:
`m_opt` can be initialized rather than set after initial construction, so it was move it before the body of the constructor.

Reply:
I suspect there should also be a dec_ref here for m_opt though. If that is the case I can add it in properly; this appears to be a referencing counting issue here.

Reply:
@NikolajBjorner should this constructor have a `dec_ref` in it as well? I assume this is a bug and not intentional? I can fix it in this PR if so.

Reply:
It shouldn't have dec-ref. The reference counting discipline is similar to the "ast" object.